subsys/net/lib/sockets/socketpair.c

```diff
@@ -0,0 +1,1132 @@
+		(*pev)->obj = &spair->write_signal;
+	}
+
+	if (pfd->events & ZSOCK_POLLOUT) {
+
+		/* Tell poll() to short-circuit wait */
+		if (!sock_is_connected(spair)) {
+			res = -EALREADY;
+			goto out;
+		}
+
+		if (*pev == pev_end) {
+			res = -ENOMEM;
+			goto out;
+		}
+
+		remote = z_get_fd_obj(spair->remote,
+			(const struct fd_op_vtable *)
+			&spair_fd_op_vtable, 0);
+
+		__ASSERT(remote != NULL, "remote is NULL");
+
+		res = k_sem_take(&remote->sem, K_FOREVER);
+		if (res < 0) {
+			goto out;
+		}
+
+		have_remote_sem = true;
+
+		/* Wait until data has been read from the remote end */
+		(*pev)->obj = &remote->read_signal;
+	}
+
+	(*pev)->type = K_POLL_TYPE_SIGNAL;
+	(*pev)->mode = K_POLL_MODE_NOTIFY_ONLY;
+	(*pev)->state = K_POLL_STATE_NOT_READY;
+	k_poll_signal_reset((*pev)->obj);
+
+	(*pev)++;
+
+	res = 0;
+
+out:
+
+	if (remote != NULL && have_remote_sem) {
+		k_sem_give(&remote->sem);
+	}
+
+	return res;
+}
+
+static int zsock_poll_update_ctx(struct spair *const spair,
+				 struct zsock_pollfd *const pfd,
+				 struct k_poll_event **pev)
+{
+	int res;
+
+	int signaled;
+	int result;
+
+	struct spair *remote = NULL;
+	bool have_remote_sem = false;
+
+	if (pfd->events & ZSOCK_POLLOUT) {
+		if (!sock_is_connected(spair)) {
+			pfd->revents |= ZSOCK_POLLHUP;
+			goto check_pollin;
+		}
+
+		remote = z_get_fd_obj(spair->remote,
+			(const struct fd_op_vtable *)
+			&spair_fd_op_vtable, 0);
+
+		__ASSERT(remote != NULL, "remote is NULL");
+
+		res = k_sem_take(&remote->sem, K_FOREVER);
+		if (res < 0) {
+			/* if other end is deleted, this might occur */
+			goto pollout_done;
+		}
+
+		have_remote_sem = true;
+
+		if (spair_write_avail(spair) > 0) {
+			pfd->revents |= ZSOCK_POLLOUT;
+			goto pollout_done;
+		}
+
+		/* check to see if op was canceled */
+		signaled = false;
+		k_poll_signal_check(
+			&remote->read_signal,
+			&signaled, &result);
+		if (signaled) {
+			/* Cannot be SPAIR_SIG_DATA, because
+			 * spair_write_avail() would have
+			 * returned 0
+			 */
+			__ASSERT(result == SPAIR_SIG_CANCEL,
+				"invalid result %d", result);
+			pfd->revents |= ZSOCK_POLLHUP;
+		}
+	}
+
+pollout_done:
+
+	if (pfd->events & ZSOCK_POLLIN) {
+		if (sock_is_eof(spair)) {
+			pfd->revents |= ZSOCK_POLLIN;
+			goto pollin_done;
+		}
+
+		if (spair_read_avail(spair) > 0) {
+			pfd->revents |= ZSOCK_POLLIN;
+			goto pollin_done;
+		}
+
+		/* check to see if op was canceled */
+		signaled = false;
+		k_poll_signal_check(
+			&spair->write_signal,
+			&signaled, &result);
+		if (signaled) {
+			/* Cannot be SPAIR_SIG_DATA, because
+			 * spair_read_avail() would have
+			 * returned 0
+			 */
+			__ASSERT(result == SPAIR_SIG_CANCEL,
+				"invalid result %d", result);
+			pfd->revents |= ZSOCK_POLLIN;
+		}
+	}
+
+pollin_done:
+	res = 0;
+
+	(*pev)++;
+
+out:
+	if (remote != NULL && have_remote_sem) {
+		k_sem_give(&remote->sem);
+	}
+
+	return res;
+}
+
+static int spair_ioctl(void *obj, unsigned int request, va_list args)
+{
+	int res;
+
```

Comment:
remove extra empty line, I would also put the shortest vars to the end of variable list, just looks better that way

Reply:
I typically put the return value first, then other uninitialized variables, then initialized variables. Will remove whitespace though.